test/UniswapAdapter.test.js

```diff
@@ -0,0 +1,80 @@
+const OracleManagerDataFeed = artifacts.require('OracleManagerDataFeed.sol')
+const UniswapAdapter = artifacts.require('UniswapAdapter.sol')
+const UniswapFactoryMock = artifacts.require('UniswapFactoryMock.sol')
+const UniswapExchangeMock = artifacts.require('UniswapExchangeMock.sol')
+const { increaseTime, uintToBytes32 } = require('./helpers')
+const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+
+contract.only('UniswapAdapter', (accounts) => {
+  let uniswapAdapter, uniswapFactoryMock, uniswapExchangeMock1, uniswapExchangeMock2
+
+  beforeEach(async () => {
+    uniswapFactoryMock = await UniswapFactoryMock.new()
+    uniswapExchangeMock1 = await UniswapExchangeMock.new(1 * 10 ** 18)
+    uniswapExchangeMock2 = await UniswapExchangeMock.new(2 * 10 ** 18)
+    uniswapAdapter = await UniswapAdapter.new(uniswapFactoryMock.address)
+  })
+
+  describe('initialize()', () => {
+    it('sets the correct UniswapFactory', async () => {
+      expect(await uniswapAdapter.uniswapFactory()).to.equal(uniswapFactoryMock.address)
+    })
+  })
+
+  describe('ping()', () => {
+    let oracleManagerDataFeed
+
+    beforeEach(async () => {
+      oracleManagerDataFeed = await OracleManagerDataFeed.new()
+      await oracleManagerDataFeed.initialize(
+        uniswapExchangeMock1.address,
+        uniswapExchangeMock2.address,
+        uniswapAdapter.address
+      )
+    })
+
+    it('updates the DataFeed with the correct price', async () => {
+      const expectedResult = uintToBytes32(await getExpectedPrice(uniswapExchangeMock1, uniswapExchangeMock2))
+      await oracleManagerDataFeed.logResult();
+      expect((await oracleManagerDataFeed.resultByIndexFor(1))[0]).to.equal(expectedResult)
```

Comment:
could also add an expect for the "date" value .. might be hard to get the exact timestamp, so maybe just a check that it's greater than 0

Reply:
👍